### ISSUE TITLE
Warnings in tests

### ISSUE THREAD

**Issue by aakashdp6548** [prob-ml/bliss#917]
There's a lot of warnings being raised in tests, primarily in `test_train_des` and `test_predict_(decals|des)`. I fixed a few in #916, but not the des/decals ones since I'm not familiar with the code.

**EDIT: a full list of warnings from the latest passing run**
```
=============================== warnings summary ===============================
tests/test_generate.py: 1 warning
tests/test_predict.py: 4 warnings
tests/test_sdss_reconstruct.py: 1 warning
tests/test_simulate.py: 3 warnings
tests/test_train.py: 3 warnings
  /home/bliss-ci/.cache/pypoetry/virtualenvs/bliss-toolkit-6tCOvDUm-py3.10/lib/python3.10/site-packages/sklearn/base.py:318: UserWarning: Trying to unpickle estimator GaussianMixture from version 1.2.1 when using version 1.2.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:
  https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations
    warnings.warn(

tests/test_predict.py::TestPredict::test_predict_decals_multiple_bricks
tests/test_predict.py::TestPredict::test_predict_decals_multiple_bricks
tests/test_predict.py::TestPredict::test_predict_des
tests/test_predict.py::TestPredict::test_predict_des
tests/test_train.py::TestTrain::test_train_des
tests/test_train.py::TestTrain::test_train_decals
  /home/bliss-ci/.cache/pypoetry/virtualenvs/bliss-toolkit-6tCOvDUm-py3.10/lib/python3.10/site-packages/astropy/wcs/wcs.py:551: FITSFixedWarning: RADECSYS= 'ICRS ' / Astrometric system 
  the RADECSYS keyword is deprecated, use RADESYSa.
    wcsprm = _wcs.Wcsprm(

tests/test_predict.py::TestPredict::test_predict_decals_multiple_bricks
tests/test_predict.py::TestPredict::test_predict_des
tests/test_train.py::TestTrain::test_train_sdss
tests/test_train.py::TestTrain::test_train_des
tests/test_train.py::TestTrain::test_train_decals
  /home/bliss-ci/.cache/pypoetry/virtualenvs/bliss-toolkit-6tCOvDUm-py3.10/lib/python3.10/site-packages/pytorch_lightning/callbacks/model_checkpoint.py:613: UserWarning: Checkpoint directory /home/bliss-ci/actions-runner/_work/bliss/bliss/checkpoints exists and is not empty.
    rank_zero_warn(f"Checkpoint directory {dirpath} exists and is not empty.")

tests/test_predict.py::TestPredict::test_predict_decals_multiple_bricks
  /home/bliss-ci/.cache/pypoetry/virtualenvs/bliss-toolkit-6tCOvDUm-py3.10/lib/python3.10/site-packages/numpy/core/fromnumeric.py:3462: RuntimeWarning: Mean of empty slice.
    return mean(axis=axis, dtype=dtype, out=out, **kwargs)

tests/test_predict.py::TestPredict::test_predict_decals_multiple_bricks
  /home/bliss-ci/.cache/pypoetry/virtualenvs/bliss-toolkit-6tCOvDUm-py3.10/lib/python3.10/site-packages/numpy/core/_methods.py:192: RuntimeWarning: invalid value encountered in divide
    ret = ret.dtype.type(ret / rcount)

tests/test_predict.py::TestPredict::test_predict_decals_multiple_bricks
  /home/bliss-ci/actions-runner/_work/bliss/bliss/bliss/surveys/decals.py:482: UserWarning: Warning: converting a masked element to nan.
    psf_params[b] = torch.tensor(np.array(band_params))

tests/test_predict.py::TestPredict::test_predict_des
tests/test_predict.py::TestPredict::test_predict_des
tests/test_train.py::TestTrain::test_train_des
  /home/bliss-ci/.cache/pypoetry/virtualenvs/bliss-toolkit-6tCOvDUm-py3.10/lib/python3.10/site-packages/astropy/wcs/wcs.py:803: FITSFixedWarning: 'datfix' made the change 'Set DATE-END to '2017-11-09T00:21:33.669' from MJD-END'.
    warnings.warn(

tests/test_predict.py::TestPredict::test_predict_des
tests/test_predict.py::TestPredict::test_predict_des
tests/test_train.py::TestTrain::test_train_des
  /home/bliss-ci/.cache/pypoetry/virtualenvs/bliss-toolkit-6tCOvDUm-py3.10/lib/python3.10/site-packages/astropy/wcs/wcs.py:803: FITSFixedWarning: 'datfix' made the change 'Set DATE-END to '2017-09-27T02:56:27.197' from MJD-END'.
    warnings.warn(

tests/test_predict.py::TestPredict::test_predict_des
tests/test_predict.py::TestPredict::test_predict_des
tests/test_train.py::TestTrain::test_train_des
  /home/bliss-ci/.cache/pypoetry/virtualenvs/bliss-toolkit-6tCOvDUm-py3.10/lib/python3.10/site-packages/astropy/wcs/wcs.py:803: FITSFixedWarning: 'datfix' made the change 'Set DATE-END to '2017-09-27T02:58:25.803' from MJD-END'.
    warnings.warn(

tests/test_predict.py::TestPredict::test_predict_des
tests/test_predict.py::TestPredict::test_predict_des
tests/test_predict.py::TestPredict::test_predict_des
tests/test_predict.py::TestPredict::test_predict_des
tests/test_train.py::TestTrain::test_train_des
tests/test_train.py::TestTrain::test_train_des
tests/test_train.py::TestTrain::test_train_des
tests/test_train.py::TestTrain::test_train_des
tests/test_train.py::TestTrain::test_train_des
  /home/bliss-ci/.cache/pypoetry/virtualenvs/bliss-toolkit-6tCOvDUm-py3.10/lib/python3.10/site-packages/skimage/restoration/deconvolution.py:[40](https://github.com/prob-ml/bliss/actions/runs/5810011182/job/15750025483#step:11:41)9: RuntimeWarning: overflow encountered in multiply
    im_deconv *= convolve(relative_blur, psf_mirror, mode='same')

tests/test_predict.py::TestPredict::test_predict_des
tests/test_predict.py::TestPredict::test_predict_des
tests/test_predict.py::TestPredict::test_predict_des
tests/test_predict.py::TestPredict::test_predict_des
tests/test_train.py::TestTrain::test_train_des
tests/test_train.py::TestTrain::test_train_des
tests/test_train.py::TestTrain::test_train_des
tests/test_train.py::TestTrain::test_train_des
tests/test_train.py::TestTrain::test_train_des
  /home/bliss-ci/.cache/pypoetry/virtualenvs/bliss-toolkit-6tCOvDUm-py3.10/lib/python3.10/site-packages/skimage/restoration/deconvolution.py:409: RuntimeWarning: invalid value encountered in multiply
    im_deconv *= convolve(relative_blur, psf_mirror, mode='same')

================= 53 passed, 53 warnings in 443.16s (0:07:23) ==================
```

**Comment by jeff-regier** [comment]
Isn't this one from code you're familiar with?
```
tests/test_encoder.py::TestEncoder::test_region_locs_in_tiles
  /home/bliss-ci/actions-runner/_work/bliss/bliss/case_studies/adaptive_tiling/region_catalog.py:32: UserWarning: Some tiles contain more than one source.
    self._validate_n_sources()
```

**Comment by jeff-regier** [comment]
I've seen this one before. @sawanp813 could you have been using a version of scikit-learn that differs from what's in `poetry.lock`?
```
  /home/bliss-ci/.cache/pypoetry/virtualenvs/bliss-toolkit-6tCOvDUm-py3.10/lib/python3.10/site-packages/sklearn/base.py:318: UserWarning: Trying to unpickle estimator GaussianMixture from version 1.2.1 when using version 1.2.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:
  https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations
    warnings.warn(
```

**Comment by jeff-regier** [comment]
Some additional warnings from the tests that aren't listed above. @zhixiangteoh Are any of these problematic? If not, would add filters for them?
```
/home/bliss-ci/.cache/pypoetry/virtualenvs/bliss-toolkit-6tCOvDUm-py3.10/lib/python3.10/site-packages/astropy/wcs/wcs.py:5[51](https://github.com/prob-ml/bliss/actions/runs/5810011182/job/15750025483?pr=916#step:11:52): FITSFixedWarning: RADECSYS= 'ICRS ' / Astrometric system 
  the RADECSYS keyword is deprecated, use RADESYSa.
    wcsprm = _wcs.Wcsprm(
```

```
tests/test_train.py::TestTrain::test_train_decals
  /home/bliss-ci/.cache/pypoetry/virtualenvs/bliss-toolkit-6tCOvDUm-py3.10/lib/python3.10/site-packages/pytorch_lightning/callbacks/model_checkpoint.py:613: UserWarning: Checkpoint directory /home/bliss-ci/actions-runner/_work/bliss/bliss/checkpoints exists and is not empty.
    rank_zero_warn(f"Checkpoint directory {dirpath} exists and is not empty.")
```

```
tests/test_predict.py::TestPredict::test_predict_decals_multiple_bricks
  /home/bliss-ci/.cache/pypoetry/virtualenvs/bliss-toolkit-6tCOvDUm-py3.10/lib/python3.10/site-packages/numpy/core/fromnumeric.py:3462: RuntimeWarning: Mean of empty slice.
    return mean(axis=axis, dtype=dtype, out=out, **kwargs)
```

```
tests/test_predict.py::TestPredict::test_predict_decals_multiple_bricks
  /home/bliss-ci/.cache/pypoetry/virtualenvs/bliss-toolkit-6tCOvDUm-py3.10/lib/python3.10/site-packages/numpy/core/_methods.py:192: RuntimeWarning: invalid value encountered in divide
    ret = ret.dtype.type(ret / rcount)
```

```
tests/test_predict.py::TestPredict::test_predict_decals_multiple_bricks
  /home/bliss-ci/actions-runner/_work/bliss/bliss/bliss/surveys/decals.py:482: UserWarning: Warning: converting a masked element to nan.
    psf_params[b] = torch.tensor(np.array(band_params))
```

```
tests/test_train.py::TestTrain::test_train_des
  /home/bliss-ci/.cache/pypoetry/virtualenvs/bliss-toolkit-6tCOvDUm-py3.10/lib/python3.10/site-packages/astropy/wcs/wcs.py:803: FITSFixedWarning: 'datfix' made the change 'Set DATE-END to '2017-09-27T02:[58](https://github.com/prob-ml/bliss/actions/runs/5810011182/job/15750025483?pr=916#step:11:59):25.803' from MJD-END'.
    warnings.warn(
```

tests/test_train.py::TestTrain::test_train_des
  /home/bliss-ci/.cache/pypoetry/virtualenvs/bliss-toolkit-6tCOvDUm-py3.10/lib/python3.10/site-packages/skimage/restoration/deconvolution.py:409: RuntimeWarning: overflow encountered in multiply
    im_deconv *= convolve(relative_blur, psf_mirror, mode='same')
```

```
tests/test_train.py::TestTrain::test_train_des
  /home/bliss-ci/.cache/pypoetry/virtualenvs/bliss-toolkit-6tCOvDUm-py3.10/lib/python3.10/site-packages/skimage/restoration/deconvolution.py:409: RuntimeWarning: invalid value encountered in multiply
    im_deconv *= convolve(relative_blur, psf_mirror, mode='same')
```

**Comment by sawanp813** [comment]
> 

My pip has v1.2.2 which is what's in our repo `poetry.lock`. I guess it's possible I was using a different environment but I don't think I ever have.

**Comment by aakashdp6548** [comment]
> Even if you were using 1.2.1, it's annoying that scikit complains about it. (Why would they making breaking changes between 1.2.1 and 1.2.2?) Is there an alternate way to store GMM parameters that's more robust to slight changes to scikit version?

Scikit suggests using `skops` instead of `pickle` to save models, but that may be overkill - we could just suppress the warning if we're not too concerned about this:
```
with warnings.catch_warnings():
    warnings.filter_warnings("ignore")
    star_gmm = pickle.load(...)
    ...
```

**Comment by zhixiangteoh** [comment]
> Some additional warnings from the tests that aren't listed above. @zhixiangteoh Are any of these problematic? If not, would add filters for them?
> 
> ```
> /home/bliss-ci/.cache/pypoetry/virtualenvs/bliss-toolkit-6tCOvDUm-py3.10/lib/python3.10/site-packages/astropy/wcs/wcs.py:5[51](https://github.com/prob-ml/bliss/actions/runs/5810011182/job/15750025483?pr=916#step:11:52): FITSFixedWarning: RADECSYS= 'ICRS ' / Astrometric system 
>   the RADECSYS keyword is deprecated, use RADESYSa.
>     wcsprm = _wcs.Wcsprm(
> ```
> 
> ```
> tests/test_train.py::TestTrain::test_train_decals
>   /home/bliss-ci/.cache/pypoetry/virtualenvs/bliss-toolkit-6tCOvDUm-py3.10/lib/python3.10/site-packages/pytorch_lightning/callbacks/model_checkpoint.py:613: UserWarning: Checkpoint directory /home/bliss-ci/actions-runner/_work/bliss/bliss/checkpoints exists and is not empty.
>     rank_zero_warn(f"Checkpoint directory {dirpath} exists and is not empty.")
> ```
> 
> ```
> tests/test_predict.py::TestPredict::test_predict_decals_multiple_bricks
>   /home/bliss-ci/.cache/pypoetry/virtualenvs/bliss-toolkit-6tCOvDUm-py3.10/lib/python3.10/site-packages/numpy/core/fromnumeric.py:3462: RuntimeWarning: Mean of empty slice.
>     return mean(axis=axis, dtype=dtype, out=out, **kwargs)
> ```
> 
> ```
> tests/test_predict.py::TestPredict::test_predict_decals_multiple_bricks
>   /home/bliss-ci/.cache/pypoetry/virtualenvs/bliss-toolkit-6tCOvDUm-py3.10/lib/python3.10/site-packages/numpy/core/_methods.py:192: RuntimeWarning: invalid value encountered in divide
>     ret = ret.dtype.type(ret / rcount)
> ```
> 
> ```
> tests/test_predict.py::TestPredict::test_predict_decals_multiple_bricks
>   /home/bliss-ci/actions-runner/_work/bliss/bliss/bliss/surveys/decals.py:482: UserWarning: Warning: converting a masked element to nan.
>     psf_params[b] = torch.tensor(np.array(band_params))
> ```
> 
> ```
> tests/test_train.py::TestTrain::test_train_des
>   /home/bliss-ci/.cache/pypoetry/virtualenvs/bliss-toolkit-6tCOvDUm-py3.10/lib/python3.10/site-packages/astropy/wcs/wcs.py:803: FITSFixedWarning: 'datfix' made the change 'Set DATE-END to '2017-09-27T02:[58](https://github.com/prob-ml/bliss/actions/runs/5810011182/job/15750025483?pr=916#step:11:59):25.803' from MJD-END'.
>     warnings.warn(
> ```
> 
> ```
> tests/test_train.py::TestTrain::test_train_des
>   /home/bliss-ci/.cache/pypoetry/virtualenvs/bliss-toolkit-6tCOvDUm-py3.10/lib/python3.10/site-packages/skimage/restoration/deconvolution.py:409: RuntimeWarning: overflow encountered in multiply
>     im_deconv *= convolve(relative_blur, psf_mirror, mode='same')
> ```

Yup some of these, like the `FITSFixedWarning` ones, are similar to #156; will silence. Will look into the others.

**Comment by jeff-regier** [comment]
Just 4 warnings left:
```
tests/test_train.py: 3 warnings
  /home/bliss-ci/.cache/pypoetry/virtualenvs/bliss-toolkit-6tCOvDUm-py3.10/lib/python3.10/site-packages/sklearn/base.py:318: UserWarning: Trying to unpickle estimator GaussianMixture from version 1.2.1 when using version 1.2.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:
  https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations
    warnings.warn(

tests/test_predict.py::TestPredict::test_predict_decals_multiple_bricks
  /home/bliss-ci/.cache/pypoetry/virtualenvs/bliss-toolkit-6tCOvDUm-py3.10/lib/python3.10/site-packages/numpy/core/fromnumeric.py:3462: RuntimeWarning: Mean of empty slice.
    return mean(axis=axis, dtype=dtype, out=out, **kwargs)

tests/test_predict.py::TestPredict::test_predict_decals_multiple_bricks
  /home/bliss-ci/.cache/pypoetry/virtualenvs/bliss-toolkit-6tCOvDUm-py3.10/lib/python3.10/site-packages/numpy/core/_methods.py:192: RuntimeWarning: invalid value encountered in divide
    ret = ret.dtype.type(ret / rcount)

tests/test_predict.py::TestPredict::test_predict_decals_multiple_bricks
  /home/bliss-ci/actions-runner/_work/bliss/bliss/bliss/surveys/decals.py:[48](https://github.com/prob-ml/bliss/actions/runs/5813434964/job/15763700841#step:12:49)2: UserWarning: Warning: converting a masked element to nan.
    psf_params[b] = torch.tensor(np.array(band_params))
```

**Comment by aakashdp6548** [comment]
The scikit model warnings from `test_train` are fixed in #921, the only ones left are the `test_predict` ones now.

**Comment by jeff-regier** [comment]
The last 3 warnings may have a common cause. They're all from `test_predict_decals_multiple_bricks` and could be due to taking a mean of tensor that's empty/masked.
